enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,411 @@
+For each edge declared by a release image (primary metadata), the graph-builder will check the blocker cache for matching blocks.
+Edges with no matching blocks are unconditionally recommended, and will be included in `edges`.
+Edges with matching blocks are conditionally recommended, and will be included in `conditionalEdges`.
+
+### Cluster-version operator support for the enhanced schema
+
+The cluster-version operator will learn to parse [`conditionalEdges`](#enhanced-cincinnati-json-representation) into [`conditionalUpdates`](#enhanced-clusterversion-representation).
+`edges` will continue to go straight into `availableUpdates`.
+The operator will log an error if the same target is included in both `edges` and `conditionalEdges`, but will prefer the `conditionalEdges` entry in that case.
+FIXME: maybe restructure `conditionalEdges` so duplicate entries are impossible?
+
+Additionally, the operator will continually re-evaluate the blocking conditionals in `conditionalUpdates` and update `conditionalUpdates[].conditions` accordingly.
+The timing of the evaluation and fresheness are largely internal details.
+The operator can periodically poll, or use edge triggered watches, combinations of watching and polling, etc.
+FIXME: [Do we need to report on recommendation freshness?](#reporting-recommendation-freshness)
+
+To perform the PromQL request, the operator will FIXME: details about connecting to the local Thanos/Prometheus.
+
+FIXME: [`platform` or other properties that are easier to evaluate than PromQL?](#non-promql-filters)
+
+If there are issues evaluating a conditional update, the operator will set the `Unknown` status on the `Recommended` condition.
+The operator will grow a new `warning`-level `CannotEvaluateConditionalUpdates` alert that fires if `lastTransitionTime` for any `Recommended=Unknown` condition is over an hour old.
+
+Any `conditionalUpdates` with `Recommended=True` will have its release inserted into `availableUpdates`.
+
+Both `availableUpdates` and `conditionalUpdates` should be sorted in decreasing SemVer order for stability, to avoid unnecessary status churn.
+
+### Update client support for the enhanced schema
+
+[The web-console][web-console] and [`oc`][oc] both consume ClusterVersion to present clients with a list of available updates.
+With this enhancement, they will both be extended to consume [`conditionalUpdates`](#enhanced-clusterversion-representation).
+When listing recommended updates, clients will list the contents of `availableUpdates`.
+When listing all supported updates, clients will additionally include entries from `conditionalUpdates` with `Recommended!=True`, and include the `reason` and `message` from the `Recommended` condition alongside the supported-but-not-recommended updates.
+Clients may optionally provision for reporting additional condition types, in case new types are added in the future.
+
+### User Stories
```

Comment:
We (engineering or at least the OTA team) should start being consistent here. If we're not going to structure these and present them as true `user stories` then we should just give the section a more descriptive heading and leave `User Stories` empty, N/A or whatever. I got dinged on this in one my enhancements but my motivation is also that I agree that user stories should include user stores.

Reply:
I think this is a good observation.

- As a cincinatti-graph-data maintainer I need a method to express targetted edge blocks, so that I can allow clusters unaffected by well scoped bugs to continue receiving the latest features or bug-fixes while protecting those clusters which would be affected by the defect (This is particularly useful during the time immediately after GA where there's tension between blocking all minor version upgrades due to issues that can be scoped to a subset of clusters.)
- As an admin of a cluster unaffected by a targetted edge block I want to upgrade as I always have
- As an admin of a cluster which has matched the criteria for a targetted edge block I do not want to update to the affected version without confirmation
- As an admin of a cluster which has matched the criteria for one or more targetted edges I want to review those upgrades and the match criteria
- As an admin of a cluster which matches targetted edge block criteria I want to upgrade to an affected version with an override specific to targetted edge blocks
- As a support engineer I want the clusterversion history to note the evaluated status of all matching criteria whenever an upgrade to a version with targetted edge blocks was taken, so that I know either the CVO did not match any criteria or the admin override the CVO.

Reply:
Ok, I've pushed 73569bc -> e417d99 to move all of the worked examples into the user stories section.  Hopefully that addresses this general point?  I'll optimistically mark this resolved, but feel free to re-open or add more specific comments if it's missing something.